dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -151,6 +151,49 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
     */
   def project[U](implicit projector: SmartProject[T,U]): TypedDataset[U] = projector.apply(this)
 
+  /** Returns a new [[TypedDataset]] that contains the elements of both this and the `other` [[TypedDataset]]
+    * combined.
+    *
+    * Note that, this function is not a typical set union operation, in that it does not eliminate
+    * duplicate items. As such, it is analogous to `UNION ALL` in SQL.
+    *
+    * Differs from `Dataset#union` by aligning fields if possible.
+    * It will not compile if `Datasets` have not compatible schema.
+    *
+    * Example:
+    * {{{
+    *   case class Foo(x: Int, y: Long)
+    *   case class Bar(y: Long, x: Int)
+    *   case class Faz(x: Int, y: Int, z: Int)
+    *
+    *   foo: TypedDataset[Foo] = ...
+    *   bar: TypedDataset[Bar] = ...
+    *   faz: TypedDataset[Faz] = ...
+    *
+    *   foo union bar: TypedDataset[Foo]
+    *   foo union faz: TypedDataset[Foo]
+    *   // won't compile, you need to reverse order, you can't project from more fields to less
```

Comment:
I think this wording is more correct: "can't project from less fields to more"

Reply:
👍